### ISSUE TITLE
Duplicate tags ending up in openapi.json, causing ReDoc issue [fixed, with PR]

### ISSUE THREAD

**Issue by aradipe** [tiangolo/fastapi#5534]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter, FastAPI, Request
from fastapi.responses import JSONResponse

app = FastAPI()
router = APIRouter(tags=["items", "items"])

@router.get("/items/")
def read_items(request: Request):
    return JSONResponse({"hello": "world"})

app.include_router(router)

# or just simply
@app.get('/test/', tags=["test", "test"])
def read_test():
    return "test"
```

### Description

* Open the browser and call endpoint '/redoc'
* Observe duplicate entries for /items and /test:

![Screenshot 2022-10-24 at 11 23 31](https://user-images.githubusercontent.com/58346970/197496996-4aba111c-8438-4d6e-b0c5-e27b1977f7b5.png)

This is caused by treating tags as a List of strings or enum inside APIRouter.
The root cause is duplicate tags ending up in openapi.json:

![Screenshot 2022-10-24 at 11 22 35](https://user-images.githubusercontent.com/58346970/197497176-b4201979-7b44-4459-aeec-287a855fd913.png)

Using Class Based Views in fastapi-utils trigger this bug (external package).

Added a fix for the issue in PR #5533 (with test)

### Operating System

macOS

### FastAPI Version

0.85.1

### Python Version

Python 3.10.8

**Comment by JarroVGIT** [comment]
Just answering here as well (copy paste from PR comment):

If you provide the same tag twice, it ends up twice in the openapi.json. That doesn't seem like that is wrong behaviour at all, nor isn't it against the spec (it is [against the spec](https://swagger.io/specification/#openapi-object) to list the same tag twice in the openapi document tags key). That said, SwaggerUI plays nicely along (as it should) and only redocs seems to have an issue with it. I would argue this need to be solved in Redoc, not in FastAPI.

**Comment by aradipe** [comment]
@JarroVGIT Thanks for taking a look and sorry for the late reply!
I've checked the openapi specs and indeed, only the global tags are restricted to be unique. To be perfectly honest I can't think of a usecase where you would want to specify duplicate tags for an API operation ("Operation Object" in spec), but that is irrelevant here.
I will probably report the tag duplication with CBV at [fastapi-utils](https://github.com/dmontagu/fastapi-utils/) as I don't think it is intentional and also an issue with ReDoc.
In the meantime I'll double-check whether FastApi allows for global (OpenAPIObject) tag duplication as that would indeed violate the spec.

**Comment by aradipe** [comment]
> In the meantime I'll double-check whether FastApi allows for global (OpenAPIObject) tag duplication as that would indeed violate the spec.
Yes, it does, I've opened issue #5552 and I'm closing this and PR #5533.
Thanks!

**Comment by tiangolo** [comment]
Thanks for the help here @JarroVGIT ! ☕

And thanks @aradipe for coming back to close the issue. 🍰
